sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredential.java

```diff
@@ -50,6 +50,22 @@
                     configuration.get(Configuration.PROPERTY_AZURE_CLIENT_ID),
                     configuration.get(Configuration.PROPERTY_AZURE_CLIENT_SECRET),
                     identityClientOptions);
+            } else if (configuration.contains(Configuration.PROPERTY_AZURE_CLIENT_ID)
```

Comment:
Seems like we should introduce a method something like envContains(params string var) that returns true if the env contains all of the 'var's.